http/src/main.rs

```diff
@@ -0,0 +1,515 @@
+use apollo_query_planner::model::Selection::Field;
+use apollo_query_planner::model::Selection::InlineFragment;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::RwLock;
+use tide::{http::Method, Body, Request, Response, StatusCode};
+
+use apollo_query_planner::model::*;
+use apollo_query_planner::QueryPlanner;
+use futures::future::{BoxFuture, FutureExt};
+use graphql_parser::schema;
+
+#[async_std::main]
+async fn main() -> std::result::Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    tide::log::start();
+    let mut app = tide::new();
+
+    // Start server
+    app.at("/").post(|req: Request<()>| async move {
+        let request_context = req.body_graphql().await?;
```

Comment:
:O

Reply:
The quality contributions Avery is known for. Keep up the good work!